lib/ssl-test.rb

```diff
@@ -56,4 +64,135 @@ def self.matching_domains domains, hostname
     domains.map {|s| Regexp.new("\A#{Regexp.escape(s).gsub('\*', '[^.]+')}\z") }
       .select {|domain| domain.match?(hostname) }
   end
+
+  # https://docs.ruby-lang.org/en/2.2.0/OpenSSL/OCSP.html
+  # https://stackoverflow.com/questions/16244084/how-to-programmatically-check-if-a-certificate-has-been-revoked#answer-16257470
+  # Returns an array with [ocsp_check_failed, certificate_revoked, error_reason, revocation_date]
+  def self.test_ocsp_revocation cert, chain, open_timeout: 5, read_timeout: 5, redirection_limit: 5
+    @ocsp_response_cache ||= {}
```

Comment:
Not thread safe (same later on), but in the worst case scenario we'd have to perform the request several times when we could have avoided it. It's not a problem for the integrity. Not a big deal IMO.

Reply:
Yeah it's more than enough ;)